psydac/fem/vector.py

```diff
@@ -351,27 +370,22 @@ def __str__(self):
         txt += '> nbasis :: ({dims})\n'.format(dims=dims)
         return txt
 
+
 #===============================================================================
-class ProductFemSpace( FemSpace ):
+class MultipatchFemSpace( FemSpace ):
     """
-    Product of FEM spaces
-    this class is used to represent FEM spaces on a multi-patch domain.
+    Product of single-patch FEM spaces
     """
 
-    def __new__(cls, *spaces, connectivity=None):
-
-        if len(spaces) == 1:
-            return spaces[0]
-        else:
-            return FemSpace.__new__(cls)
-
-    def __init__( self, *spaces, connectivity=None):
+    def __init__( self, *spaces, connectivity={}):
```

Comment:
It is dangerous to pass a mutable object (like a dictionary) as a default value to a function, because this gets evaluated when the function is defined. Here I would leave the original `None`.

Reply:
ok, good point!

Reply:
Done in 6490fa3